Userland/Libraries/LibJS/Tests/eval-basic.js

```diff
@@ -30,3 +30,41 @@ test("returns 1st argument unless 1st argument is a string", () => {
     var stringObject = new String("1 + 2");
     expect(eval(stringObject)).toBe(stringObject);
 });
+var value="outer"
+test("only touches locals if direct access", function() {
+    var value = "inner"
+    expect(global.eval("value")).toEqual("outer");
+});
+
+test("eval pointing to another function calls function", function() {
+    var value = "inner"
+    var eval = ()=>{return "wat"}
+    expect(eval("value")).toEqual("wat");
+});
+
+test("eval pointing to eval works as a direct eval", function() {
+    var value = "inner"
+    var eval = global.eval
+    expect(eval("value")).toEqual("inner");
+});
+
+test("alias to eval works as a global eval", function() {
+    var value = "inner"
+    var eval1 = global.eval
+    expect(eval1("value")).toEqual("outer");
+});
+test("eval evaluates all args", function() {
+    var i = 0
+    expect(eval("value", i++, i++, i++)).toEqual("outer");
+    expect(i).toEqual(3)
+});
+
+test("eval tests for exceptions", function() {
+    var i = 0
+    try {
+        eval("value", i++, i++, j, i++)
+        expect.unreached();
```

Comment:
`expect.unreached` isn't a function. You could do `expect.fail('message')`, however the test suite relies on exceptions to communicate failure, so you'd catch the exception being thrown anyways. A better way to do this would be something along the lines of:

```js
expect(() => {
    eval("value", i++, i++, j, i++);
}).toThrow();
expect(i).toEqual(2);
```

And depending on what it throws (should be a `ReferenceError`, right?), you could also do something like `.toThrowWithMessage(ReferenceError, "value isn't defined or whatever the message is")`

Reply:
Dur, I meant to make that test actually correct, will fix that :D

Reply:
Corrected this I think